DSharpPlus/Entities/Guild/DiscordMember.cs

```diff
@@ -587,6 +587,45 @@ public async Task UpdateVoiceStateAsync(DiscordChannel channel, bool? suppress)
         public Permissions PermissionsIn(DiscordChannel channel)
             => channel.PermissionsFor(this);
 
+        /// <summary>
+        /// Constructs the url for a guild member's avatar, defaulting to the user's avatar if none is set.
+        /// </summary>
+        /// <param name="imageFormat">The image format of the avatar to get.</param>
+        /// <param name="imageSize">The maximum size of the avatar. Must be a power of two, minimum 16, maximum 4096.</param>
+        /// <returns>The URL of the user's avatar.</returns>
+        public string GetGuildAvatarUrl(ImageFormat imageFormat, ushort imageSize = 1024)
+        {
+            // Run this if statement before any others to prevent running the if statements twice.
+            if (string.IsNullOrWhiteSpace(this.AvatarHash))
+            {
+                return this.GetAvatarUrl(imageFormat, imageSize);
+            }
+
+            if (imageFormat == ImageFormat.Unknown)
+                throw new ArgumentException("You must specify valid image format.", nameof(imageFormat));
+
+            // Makes sure the image size is in between Discord's allowed range.
+            if (imageSize < 16 || imageSize > 4096)
+                throw new ArgumentOutOfRangeException("Image Size is not in between 16 and 4096: " + nameof(imageSize));
+
+            // Checks to see if the image size is a power of two.
+            if (imageSize is 0 && (imageSize & (imageSize - 1)) is not 0)
```

Comment:
I did say `if (x is *not* 0`

Reply:
Yea, we doing checks to make sure it is a power of two, and throw an exception if it isn't. Misleading comment, will fix

Reply:
No, the if check is wrong. As discussed on Discord I showed you this: `x is not 0 && x & (x - 1) is 0`. The `is` checks are backward.

Reply:
This is still broken. by changing `is not 0` to `is 0`, you've created an unreachable condition (which the compiler may or may not warn you about), and this is never true, so an exception is never thrown. Furthermore, by inverting the checks, you've broken the entire computation. What should be done instead is wrapping this in parentheses and inverting the result of that.

Reply:
Should be fixed on latest commit